resourcescaler.go

```diff
@@ -113,82 +235,244 @@ func (s *AppResourceScaler) patchIguazioTenantAppServiceSets(namespace string, j
+	if err != nil {
+		return "", time.Now(), errors.Wrap(err, "Failed to parse scale state")
 	}
 
-	s.logger.DebugWith("Found services", "services", resources)
+	lastScaleStateTimeString, ok := scaleToZeroStatus["last_scale_state_time"].(string)
+	if !ok {
+		return "", time.Now(), errors.New("Scale to zero status does not have last scale state time")
+	}
 
-	return resources, nil
+	lastScaleStateTime, err := time.Parse(time.RFC3339, lastScaleStateTimeString)
+	if err != nil {
+		return "", time.Now(), errors.Wrap(err, "Failed to parse last scale state time")
+	}
+
+	return lastScaleState, lastScaleStateTime, nil
 }
 
-func (s *AppResourceScaler) GetConfig() (*scaler_types.ResourceScalerConfig, error) {
-	return nil, nil
+func (s *AppResourceScaler) parseLastScaleState(serviceStatus interface{}) (scaler_types.ScaleState, time.Time, error) {
+	serviceStatusMap, ok := serviceStatus.(map[string]interface{})
+	if !ok {
+		return "", time.Now(), errors.New("Service status type assertion failed")
+	}
+
+	scaleToZeroStatus, ok := serviceStatusMap["scale_to_zero"].(map[string]interface{})
+	if !ok {
+		s.logger.DebugWith("Service does not have scale to zero status (it is ok), using defaults")
+		return scaler_types.NonScaleState, time.Now(), nil
+	}
+
+	lastScaleState, lastScaleStateTime, err := s.parseScaleToZeroStatus(scaleToZeroStatus)
+	if err != nil {
+		return "", time.Now(), errors.Wrap(err, "Failed parsing scale to zero status")
+	}
+
+	return lastScaleState, lastScaleStateTime, nil
 }
 
-func getClientConfig(kubeconfigPath string) (*rest.Config, error) {
-	if kubeconfigPath != "" {
-		return clientcmd.BuildConfigFromFlags("", kubeconfigPath)
+func (s *AppResourceScaler) parseServiceState(serviceStatus interface{}) (string, error) {
+	serviceStatusMap, ok := serviceStatus.(map[string]interface{})
+	if !ok {
+		return "", errors.New("Service status type assertion failed")
 	}
 
-	return rest.InClusterConfig()
+	stateString, ok := serviceStatusMap["state"].(string)
+	if !ok {
+		return "", errors.New("Service status does not have state")
+	}
+
+	return stateString, nil
+}
+
+func (s *AppResourceScaler) parseScaleResources(serviceSpecInterface interface{}) ([]scaler_types.ScaleResource, error) {
+	var parsedScaleResources []scaler_types.ScaleResource
+	serviceSpec, ok := serviceSpecInterface.(map[string]interface{})
+	if !ok {
+		return []scaler_types.ScaleResource{}, errors.New("Service spec type assertion failed")
+	}
+
+	scaleToZeroSpec, ok := serviceSpec["scale_to_zero"].(map[string]interface{})
+	if !ok {
+
+		// It's ok for a service to not have the scale_to_zero spec
+		return []scaler_types.ScaleResource{}, nil
+	}
+
+	scaleToZeroMode, ok := scaleToZeroSpec["mode"].(string)
+	if !ok {
+		return []scaler_types.ScaleResource{}, errors.New("Scale to zero spec does not have mode")
+	}
+
+	// if it's not enabled there's no reason to parse the rest
+	if scaleToZeroMode == "enabled" {
```

Comment:
Can be nice to check if `!= 'enabled'` and return early, then the rest of this won't be indented